src/daft-core/src/array/fixed_size_list_array.rs

```diff
@@ -198,8 +198,18 @@ impl FixedSizeListArray {
     }
 
     pub fn to_list(&self) -> ListArray {
+        let field = self.field.clone();
+
+        let DataType::FixedSizeList(inner_type, _) = &field.dtype else {
+            panic!("Expected FixedSizeListArray, got {:?}", field.dtype);
```

Comment:
I think this should be `unreachable!` as this should be infallible at this point.

Reply:
1a2ba6f496c4ad2e0a1fbf7262e07b249a4ad117